#! /bin/sh /usr/share/dpatch/dpatch-run
## 10_gcc-4.3-fixes.dpatch by  <ansgar@43-1.org>
## Refreshed by  <brad@brad-smith.co.uk>
##
## DP: Fix compliation with gcc-4.3

@DPATCH@

diff -Naur briquolo-0.5.7.orig/src/Constante.cpp briquolo-0.5.7/src/Constante.cpp
--- briquolo-0.5.7.orig/src/Constante.cpp	2008-03-22 09:40:33.000000000 +0000
+++ briquolo-0.5.7/src/Constante.cpp	2008-03-29 18:47:44.000000000 +0000
@@ -20,6 +20,7 @@
  *
  *****************************************************************************/
 #include "Constante.h"
+#include <cstdlib>
 
 #ifdef HAVE_CONFIG_H
 #include "config.h"
diff -Naur briquolo-0.5.7.orig/src/I18n.cpp briquolo-0.5.7/src/I18n.cpp
--- briquolo-0.5.7.orig/src/I18n.cpp	2006-03-07 20:31:24.000000000 +0000
+++ briquolo-0.5.7/src/I18n.cpp	2008-03-29 18:47:44.000000000 +0000
@@ -22,6 +22,7 @@
 #include "I18n.h"
 #include <iostream>
 #include "Constante.h"
+#include <cstdlib>
 
 #ifdef WIN32
   #include <windows.h>
diff -Naur briquolo-0.5.7.orig/src/MOGL/MOGL_EnsembleObjet.cpp briquolo-0.5.7/src/MOGL/MOGL_EnsembleObjet.cpp
--- briquolo-0.5.7.orig/src/MOGL/MOGL_EnsembleObjet.cpp	2008-03-24 13:12:36.000000000 +0000
+++ briquolo-0.5.7/src/MOGL/MOGL_EnsembleObjet.cpp	2008-03-29 18:47:44.000000000 +0000
@@ -24,7 +24,8 @@
 #include "MOGL_Armature.h"
 #include "MOGL_Peau.h"
 //#include
-#include <stdio.h>
+#include <cstdio>
+#include <cstring>
 
 //#include <mmsystem.h>
 
diff -Naur briquolo-0.5.7.orig/src/MOGL/MOGL_GestionnaireObjet.cpp briquolo-0.5.7/src/MOGL/MOGL_GestionnaireObjet.cpp
--- briquolo-0.5.7.orig/src/MOGL/MOGL_GestionnaireObjet.cpp	2008-03-24 12:56:13.000000000 +0000
+++ briquolo-0.5.7/src/MOGL/MOGL_GestionnaireObjet.cpp	2008-03-29 18:47:44.000000000 +0000
@@ -21,6 +21,7 @@
  *****************************************************************************/
 #include "../I18n.h"
 #include "MOGL_GestionnaireObjet.h"
+#include <cstring>
 
 bool MOGL_GestionnaireObjet::ChargerObjetASCTriangle(const char * p_NomFichier, const MOGL_GestionnaireTexture & p_GM,
                                                      MOGL_Objet & p_Objet)
diff -Naur briquolo-0.5.7.orig/src/MOGL/MOGL_GestionnaireObjet.h briquolo-0.5.7/src/MOGL/MOGL_GestionnaireObjet.h
--- briquolo-0.5.7.orig/src/MOGL/MOGL_GestionnaireObjet.h	2006-03-07 20:31:24.000000000 +0000
+++ briquolo-0.5.7/src/MOGL/MOGL_GestionnaireObjet.h	2008-03-29 18:47:44.000000000 +0000
@@ -24,7 +24,7 @@
 
 #include <vector>
 #include <iostream>
-#include <stdio.h>
+#include <cstdio>
 
 #include "MOGL_Triangle.h"
 #include "MOGL_Objet.h"
diff -Naur briquolo-0.5.7.orig/src/MOGL/MOGL_Objet.cpp briquolo-0.5.7/src/MOGL/MOGL_Objet.cpp
--- briquolo-0.5.7.orig/src/MOGL/MOGL_Objet.cpp	2006-03-07 20:31:24.000000000 +0000
+++ briquolo-0.5.7/src/MOGL/MOGL_Objet.cpp	2008-03-29 18:47:44.000000000 +0000
@@ -20,6 +20,7 @@
  *
  *****************************************************************************/
 #include "MOGL_Objet.h"
+#include <cstdlib>
 
 //MOGL_Objet::MOGL_Objet(): MOGL_ObjetAbstrait(), MOGL_ElementArbre()
 MOGL_Objet::MOGL_Objet(): MOGL_Noeud(), _ModifieTriangle(true), _ModifieStrip(true),
diff -Naur briquolo-0.5.7.orig/src/MOGL/MOGL_Peau.cpp briquolo-0.5.7/src/MOGL/MOGL_Peau.cpp
--- briquolo-0.5.7.orig/src/MOGL/MOGL_Peau.cpp	2006-03-07 20:31:24.000000000 +0000
+++ briquolo-0.5.7/src/MOGL/MOGL_Peau.cpp	2008-03-29 18:47:44.000000000 +0000
@@ -20,6 +20,7 @@
  *
  *****************************************************************************/
 #include "MOGL_Peau.h"
+#include <cstdlib>
 
 MOGL_Peau::MOGL_Peau(): MOGL_Noeud(), _ModifieTriangle(true)
 {
